sycl/test/host-interop-task/host-task-dependency.cpp

```diff
@@ -0,0 +1,201 @@
+      for (size_t Idx = 0; Idx < GeneratorAcc.get_count(); ++Idx)
+        GeneratorAcc[Idx] = Idx;
+    };
+
+    CGH.single_task<class GeneratorTask>(GeneratorKernel);
+  });
+
+  // 2. submit host task writing from buf 1 to buf 2
+  auto HostTaskEvent = Ctx.Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::host_buffer>
+        CopierSrcAcc(Ctx.Buf1, CGH);
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::host_buffer>
+        CopierDstAcc(Ctx.Buf2, CGH);
+
+    auto CopierHostTask = [CopierSrcAcc, CopierDstAcc, &Ctx]() {
+      for (size_t Idx = 0; Idx < CopierDstAcc.get_count(); ++Idx)
+        CopierDstAcc[Idx] = CopierSrcAcc[Idx];
+
+      bool Expected = false;
+      bool Desired = true;
+      assert(Ctx.Flag.compare_exchange_strong(Expected, Desired));
+
+      // let's employ some locking here
+      {
+        std::lock_guard<std::mutex> Lock(Ctx.Mutex);
+        Ctx.CV.notify_all();
+      }
+    };
+
+    CGH.codeplay_host_task(CopierHostTask);
+  });
+
+  // 3. submit simple task to move data between two buffers
+  Ctx.Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::global_buffer>
+        SrcAcc(Ctx.Buf2, CGH);
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::global_buffer>
+        DstAcc(Ctx.Buf3, CGH);
+
+    CGH.depends_on(HostTaskEvent);
+
+    auto CopierKernel = [SrcAcc, DstAcc]() {
+      for (size_t Idx = 0; Idx < DstAcc.get_count(); ++Idx)
+        DstAcc[Idx] = SrcAcc[Idx];
+    };
+
+    CGH.single_task<class CopierTask>(CopierKernel);
+  });
+
+  // 4. check data in buffer #3
+  {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::host_buffer>
+        Acc(Ctx.Buf3);
+
+    bool Failure = false;
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx) {
+      fprintf(stderr, "Third buffer [%3zu] = %i\n", Idx, Acc[Idx]);
+
+      Failure |= (Acc[Idx] != Idx);
+      //assert(Acc[Idx] == Idx && "Invalid data in third buffer");
+    }
+
+    assert(!Failure && "Invalid data in third buffer");
+  }
+}
+
+void Thread2Fn(Context &Ctx) {
+  std::unique_lock<std::mutex> Lock(Ctx.Mutex);
+
+  // T2.1. Wait until flag F is set eq true.
+  Ctx.CV.wait(Lock, [&Ctx] { return Ctx.Flag.load(); });
+
+  assert(Ctx.Flag.load());
+
+  // T2.2. print some "hello, world" message
+  Ctx.Message = "Hello, world";
+}
+
+void test() {
+  auto EH = [](S::exception_list EL) {
+    for (const std::exception_ptr &E : EL) {
+      throw E;
+    }
+  };
+
+  S::queue Queue(EH);
+
+  Context Ctx{{false}, Queue, "", {10}, {10}, {10}, {}, {}};
+
+  // 0. setup: thread 1 T1: exec smth; thread 2 T2: waits; init flag F = false
+  std::thread Thread1(Thread1Fn, std::reference_wrapper<Context>(Ctx));
+  std::thread Thread2(Thread2Fn, std::reference_wrapper<Context>(Ctx));
```

Comment:
Take a poll about the readability of this compared to the line above and give me the result :-)
```suggestion
  std::thread Thread2 { [&] { Thread2Fn(Ctx); } };
```